### PR TITLE
[Ide] Added Or and ProjectCapability file template conditions

### PR DESCRIPTION
**Pull request by DavidKarlas**
I will also open PR for Xamarin.Forms addin which will make use of this new condition

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Templates/OrFileTemplateCondition.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+using System;
+using System.Xml;
+using System.Linq;
+using MonoDevelop.Projects;
+
+namespace MonoDevelop.Ide.Templates
+{
+	public class OrFileTemplateCondition : FileTemplateCondition
```
Comment (mrward) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Templates/OrFileTemplateCondition.cs]: Does this need to be public?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Templates/ProjectCapabilityFileTemplateCondition.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+using System;
+using System.Xml;
+using MonoDevelop.Projects;
+
+namespace MonoDevelop.Ide.Templates
+{
+	public class ProjectCapabilityFileTemplateCondition : FileTemplateCondition
```
Comment (mrward) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Templates/ProjectCapabilityFileTemplateCondition.cs]: Does this need to be public?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Templates/ProjectCapabilityFileTemplateCondition.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	{
+		string capabilityExpression;
+		public override void Load (XmlElement element)
+		{
+			capabilityExpression = element.GetAttribute ("CapabilityExpression");
```
Comment (mrward) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Templates/ProjectCapabilityFileTemplateCondition.cs]: Not sure I like this attribute name. You end up using the following in the file template:

    `<ProjectCapability CapabilityExpression="XamarinForms" />`

The attribute name suggests you can write an expression? Could we just use:

`<ProjectCapability>XamarinForms</ProjectCapability>`

Or

`<ProjectCapability name="XamarinForms" />`

Comment (DavidKarlas) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Templates/ProjectCapabilityFileTemplateCondition.cs]: This was done intentionally to point that out, that user can write expression https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/Project.cs#L803-L812

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Templates/ProjectCapabilityFileTemplateCondition.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+			if (!expressionHasValue && !nameHasValue)
+				throw new InvalidOperationException ("Name or Expression must be set.");
+			if (nameHasValue && expressionHasValue)
+				throw new InvalidOperationException ("Set Name or Expression, but you can not set both at same time.");
+			useName = nameHasValue;
+		}
+
+		public override bool ShouldEnableFor (Project proj, string projectPath)
+		{
+			if (!(proj is DotNetProject dnp))
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Templates/ProjectCapabilityFileTemplateCondition.cs]: Capabilities are handled by the Project class, it doesn't need to be a DotNetProject.

**Pull request merged**